stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewTypeMapper.kt

```diff
@@ -30,14 +30,22 @@ internal object MessageListItemViewTypeMapper {
         return when {
             messageItem.message.deletedAt != null -> MessageListItemViewType.MESSAGE_DELETED
             messageItem.message.isGiphyEphemeral() -> MessageListItemViewType.GIPHY
-            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isMedia() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
-            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isNotEmpty() -> MessageListItemViewType.PLAIN_TEXT_WITH_FILE_ATTACHMENTS
+            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isMediaOrLink() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
```

Comment:
First of all, `isMediaOrLink` doesn't relate to the logic inside completely. If I understood it correctly I would call it `atLeastOneMedia`
Generally, I think all these conditions look hard to be understood. What if we made some func `Message.isMediaWithText()` and rewrite whole line as:
```kotlin
messageItem.message.isMediaWithText() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
messageItem.message.isFileWithText() -> MessageListItemViewType.PLAIN_TEXT_WITH_FILE_ATTACHMENTS
```

What do you think?

Reply:
Basically there are 3 cases:
1. Message has only media or link attachments but at least one of them is media - PLAIN_TEXT_WITH_MEDIA
2. Message has attachments and at least one of them is not link - PLAIN_TEXT_WITH_FILE
3. PLAIN_TEXT

I agree, it's quite complicated. I'll improve that